shell/testing/observatory/test.dart

```diff
@@ -57,114 +31,53 @@ Future<String> readResponse(HttpClientResponse response) {
 }
 
 // Test accessing the service protocol over http.
-Future<Null> testHttpProtocolRequest(Uri uri) async {
+Future<void> testHttpProtocolRequest(Uri uri) async {
   uri = uri.replace(path: 'getVM');
   final HttpClient client = HttpClient();
   final HttpClientRequest request = await client.getUrl(uri);
   final HttpClientResponse response = await request.close();
   Expect.equals(response.statusCode, 200);
-  final Map<String, dynamic> responseAsMap =
-      json.decode(await readResponse(response)) as Map<String, dynamic>;
+  final responseAsMap =
+      json.decode(await readResponse(response)) as Map<String, Object?>;
   Expect.equals(responseAsMap['jsonrpc'], '2.0');
   client.close();
 }
 
 // Test accessing the service protocol over ws.
-Future<Null> testWebSocketProtocolRequest(Uri uri) async {
+Future<void> testWebSocketProtocolRequest(Uri uri) async {
   uri = uri.replace(scheme: 'ws', path: 'ws');
   final WebSocket webSocketClient = await WebSocket.connect(uri.toString());
   final ServiceClient serviceClient = ServiceClient(webSocketClient);
   final Map<String, dynamic> response = await serviceClient.invokeRPC('getVM');
   Expect.equals(response['type'], 'VM');
   try {
     await serviceClient.invokeRPC('BART_SIMPSON');
-    Expect.notExecuted();
+    throw AssertionError('Unreachable');
   } catch (e) {
     // Method not found.
     Expect.equals((e as Map<String, dynamic>)['code'], -32601);
   }
 }
 
 // Test accessing an Observatory UI asset.
-Future<Null> testHttpAssetRequest(Uri uri) async {
+Future<void> testHttpAssetRequest(Uri uri) async {
   uri = uri.replace(path: 'third_party/trace_viewer_full.html');
   final HttpClient client = HttpClient();
   final HttpClientRequest request = await client.getUrl(uri);
   final HttpClientResponse response = await request.close();
   Expect.equals(response.statusCode, 200);
-  await response.drain();
+  await response.drain<void>();
   client.close();
 }
 
-Future<Null> testStartPaused(Uri uri) async {
```

Comment:
Why is this gone now?

Reply:
Dead code that's been disabled for about 6 or 7 years